editor/src/components/canvas/dom-sampler.tsx

```diff
@@ -0,0 +1,455 @@
+import type {
+  ComputedStyle,
+  DetectedLayoutSystem,
+  ElementPath,
+  StyleAttributeMetadata,
+} from 'utopia-shared/src/types'
+import {
+  fillMissingDataFromAncestors,
+  MetadataUtils,
+} from '../../core/model/element-metadata-utils'
+import { UTOPIA_PATH_KEY } from '../../core/model/utopia-constants'
+import { allElemsEqual, mapDropNulls, pluck } from '../../core/shared/array-utils'
+import { getCanvasRectangleFromElement } from '../../core/shared/dom-utils'
+import { alternativeEither, left } from '../../core/shared/either'
+import * as EP from '../../core/shared/element-path'
+import type { ElementPathTrees } from '../../core/shared/element-path-tree'
+import {
+  domElementMetadata,
+  emptySpecialSizeMeasurements,
+  type ComputedStyleMetadata,
+  type DomElementMetadata,
+  type ElementInstanceMetadata,
+  type ElementInstanceMetadataMap,
+} from '../../core/shared/element-template'
+import type { CanvasRectangle } from '../../core/shared/math-utils'
+import {
+  boundingRectangleArray,
+  infinityCanvasRectangle,
+  nullIfInfinity,
+  type CanvasPoint,
+} from '../../core/shared/math-utils'
+import { optionalMap } from '../../core/shared/optional-utils'
+import { camelCaseToDashed } from '../../core/shared/string-utils'
+import { getPathWithStringsOnDomElement } from '../../core/shared/uid-utils'
+import type { ElementsToRerender } from '../editor/store/editor-state'
+import type { CSSPosition, FlexDirection } from '../inspector/common/css-utils'
+import { computedStyleKeys } from '../inspector/common/css-utils'
+import { CanvasContainerID } from './canvas-types'
+import {
+  collectDomElementMetadataForElement,
+  getAttributesComingFromStyleSheets,
+} from './dom-walker'
+import type { UiJsxCanvasContextData } from './ui-jsx-canvas'
+
+export function runDomSampler(
+  elementsToFocusOn: ElementsToRerender,
+  options: {
```

Comment:
Why wrap these in an object?

Reply:
fixed, now all parameters are in the object. in my opinion this makes future refactors easier, and makes the caller side more readable. makes it harder to forget that you missed passing some important parameter. Case in point: I just added `domWalkerAdditionalElementsToFocusOn` and the atomic commit to do that has a very readable diff, and it made it easier for me to see where I am still yet to pass it